lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -167,6 +167,67 @@ static Value createSoftmaxBackwardCommonKernel(PatternRewriter &rewriter,
   return sub;
 }
 
+namespace {
+/// We decompose aten.amax into a set of aten.max.dim op(s) depending on the
+/// number of dimensions across which the max needs to be computed.
+/// Eg:
+///   INPUT:
+///      final_output = aten.amax(initial_input, (0, 2), keepdim=False)
+///
+///   OUTPUT:
+///      input_1 = aten.max.dim(initial_input, 0, keepdim)  #1
+///      final_output = aten.max.dim(input_1, 1, keepdim)   #2
+///
+/// NOTE: In the above example aten.max.dim #2 is sent 1 as the `dim` instead of
+///       2 because after #1 the 0-th dimension from the `initial_input` is
+///       erased.
+///       This is because `keepdim` was set as False. Had it been True, we would
+///       have sent 2 as the reducing dimension even in the second invocation of
+///       #2.
+///       Therefore for every dimension provided in aten.amax, we create a new
+///       aten.max.dim AND for every such dimension (except the first one) we'd
+///       make an adjustment (refer the code comment) before calling
+///       aten.max.dim.
+class DecomposeAtenAmaxOp : public OpRewritePattern<AtenAmaxOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenAmaxOp op,
+                                PatternRewriter &rewriter) const override {
+    Location loc = op.getLoc();
+    SmallVector<Value> dims;
+    if (!getListConstructElements(op.dim(), dims))
```

Comment:
nit: I would use a `matchPattern` to match on `m_TorchListOfConstantInts`, since you need the ints in the `for` loop. This prevents generating IR when the inputs are invalid